components/ntp_background_images/browser/sponsored_images_component_data.h

```diff
@@ -12,12 +12,19 @@
 
 namespace ntp_background_images {
 
+struct BackgroundImagesComponentData {
```

Comment:
I think we don't need to add this new struct because NTP BI uses only one component.
Just using its key and id string directly from installer seems fine.